mcf-browser/src/back-end/parse.ts

```diff
@@ -0,0 +1,184 @@
+          {
+            file: fileName,
+            errs: mcfOut['errMsgs'],
+          },
+        ]);
+      }
+
+      finalReturn['localNodes'] = finalReturn['localNodes'].concat(
+          mcfOut['localNodes'],
+      );
+    } else if (fileExt === 'csv') {
+      if (tmcfFile) {
+        const tmcfFileName = (tmcfFile as File).name;
+        const tmcfOut = await ParseTmcf.generateMcf(tmcfFile, file).then(
+            (mcf) => {
+              const mcfParser = new ParseMcf(
+                  tmcfFileName + '&' + fileName,
+              );
+              return mcfParser.parseMcfStr(mcf as string);
+            },
+        );
+
+        const datapoints = await ParseTmcf.generateDataPoints(tmcfFile, file);
+        finalReturn['datapoints'] = finalReturn['datapoints'].concat([
+          datapoints,
+        ]);
+
+        if (tmcfOut['errMsgs'].length !== 0) {
+          finalReturn['errMsgs'] = finalReturn['errMsgs'].concat({
+            file: tmcfFileName,
+            errs: tmcfOut['errMsgs'],
+          });
+        }
+        finalReturn['localNodes'] = finalReturn['localNodes'].concat(
+            tmcfOut['localNodes'],
+        );
+      }
+    }
+  }
+  return finalReturn;
+}
+
+/** Group nodes and find all time series
+  * @param {Object[]} datapoints the time series data
+  * @return {Series[]} an array of time series in the data
+  */
+function getTimeData(datapoints: Object[]) : Series[] {
+  // Turn from array of objects (one per file) to one big object
+  const allData: any = {};
+  for (const data of datapoints) {
+    const allSeries = Object.keys(data);
+    for (const series of allSeries) {
+      allData[series] = allData[series] ? allData[series] : {};
+      allData[series] = {...allData[series], ...(data as any)[series]};
+    }
+  }
+
+  // Turn from object to series
+  const output = [];
+  const allSeries = Object.keys(allData);
+  for (const series of allSeries) {
+    output.push(parseSeries(series, allData[series]));
+  }
+
+  return output;
+}
+
+/**
+  * Takes in the facet string generated when parsing the file
+  * and return an object of type Series
+  * @param {string} facet the facet defining the series
+  * @param {Object} values the values for the series
+  * @return {Series} the datapoint as a Series object
+  */
+function parseSeries(facet: string, values: Object) : Series {
+  const {
+    variableMeasured,
+    observationAbout,
+    provenance,
+    measurementMethod,
+    observationPeriod,
+    unit,
+    scalingFactor,
+  } = Series.fromID(facet);
+  const data = [];
+  for (const date of Object.keys(values)) {
+    data.push({
+      x: date,
+      y: parseFloat((values as any)[date]),
+    });
+  }
+
+  return new Series(
+      data,
+      variableMeasured,
+      observationAbout,
+      provenance,
+      measurementMethod,
+      observationPeriod,
+      unit,
+      scalingFactor,
+  );
+}
+
+export {getNodes, getTimeData, parseSeries};
```

Comment:
getTimeSeries ?

Reply:
Not sure what this is referring to, there's no method named getTimeSeries. Are you suggesting we rename getTimeData to getTimeSeries? If so, getTimeData returns the data for all of the series in a file and not just one time series so getTimeSeries might be misleading.